basic-integration/surfly-integration-with-tag-manager.md

```diff
@@ -0,0 +1,177 @@
+- The last part of the tag is Javascript. First, we've made sure that GTM will show our buttons. Then we've added the Surfly snippet, the Surfly init function, a sessionStart() function, the Olark snippet and a sessionEnd() function.
+
+```javascript
+//Buttons to control the Surfly session
+<div class="footer" id="footer">
+  <button class="basic-button" id="btn-start-session" onclick="sessionStart()"><p id="id-cover">Get Help</p></button>
+  <button class="basic-button" id="btn-end-session" onclick="sessionEnd()">Stop sessie</button>
+</div>
+
+//Button styling
+<style type="text/css">
+  .basic-button {
+    background-color: #ffd05b;
+    width: 115px;
+    border: 1px solid #ffd05b;
+    border-radius: 33px;
+    height: 50px;
+    font-size: 14px;
+    color: #333333;
+    letter-spacing: 1px;
+    font-family: sans-serif;
+    font-weight: bold;
+    position: absolute;
+    left: 15px;
+    bottom: 15px;
+    box-shadow: 2px 2px rgba(0, 0, 0, 0.1);
+    position: fixed;
+  }
+
+  .basic-button:hover {
+    cursor: pointer;
+  }
+
+  .basic-button p {
+    margin-top: 14px;
+    font-size: 13px;
+  }
+
+  #btn-end-session {
+    display: none;
+  }
+</style>
+
+<script>
+//Make sure that GTM doesn't hide the buttons
+  document.getElementById('footer').parentElement.removeAttribute('style');
+  
+//Add the Surfly snippet
+  (function(s,u,r,f,l,y){s[f]=s[f]||{init:function(){s[f].q=arguments}};
+  l=u.createElement(r);y=u.getElementsByTagName(r)[0];l.async=1;
+  l.src='https://surfly.com/surfly.js';y.parentNode.insertBefore(l,y);})
+  (window,document,'script','Surfly');
+
+// Implement the Surfly init code, ADD YOUR OWN WIDGET KEY 
+  Surfly.init({widget_key:'** add your own Surfly widget key'}, function(init) {
+    if (init.success) {
+      if (Surfly.currentSession) {
+        // Laat binnen de sessie de btn-end-session zien
+        document.getElementById('btn-end-session').style.display="block";
+        // Zorg ervoor dat de btn-start-session ruimte maakt om de session-id te laten zien
+        document.getElementById("btn-start-session").style.visibility="hidden";
+      }
+    }
+  });
+
+//Function sessionStart() that starts the Surfly session, so that it can be applied to the "Start Session"-button
+//Add options according to you wishes
+  function sessionStart() {
+    var settings = {
+      block_until_agent_joins: false,
+      hide_until_agent_joins: true,
+      start_docked: true,
+      cookie_transfer_enabled: true,
+      cookie_transfer_proxying: false,
+      splash: false
+    };
+//Code that shows the session-id so that it can be passed to the helpdesk agent
+    Surfly.session(settings)
+    .on('session_started', function(session, event) {
+      document.getElementById("id-cover").style.display = "none";
+      console.log(session);
+      var showId = document.getElementById("btn-start-session");
+      showId.style.display = "block";
+      showId.textContent = session.pin;
+    }).startLeader();
```

Comment:
Just to look even more professional, let's make the code style consistent. I would do the following (including changes for aforementioned `<p>` removal, indentation fixes, and I changed the variable names to be more relevant):

```javascript
var startButton = document.getElementById("btn-start-session");
startButton.disabled = "true";
Surfly.session(settings)
      .on('session_started', function(session) {
        startButton.textContent = session.pin;
      })
      .startLeader();
```

Note that I also change `disabled` attribute so that you can't click on the button twice. And I removed setting the display style, since the button is already visible